utils.py

```diff
@@ -0,0 +1,68 @@
+##############################################################################
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#############################################################################
+
+import pymysql.cursors
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+import os
+
+
+# Connect method used in the SQLAlchemy creator
+def connect():
+    return pymysql.connect(
+        host=os.environ['db_host'],
+        user=os.environ['db_user'],
+        password=os.environ['db_pass'],
+        db=os.environ['db_name'],
+        ssl={
+            'ssl': {
+                'ca': './ca.pem',
+                'cert': './cert.pem',
+                'key': './priv.pem'
+            }
+        }
+    )
+
+
+# Helper function to generate the private and public keys
+def generate_keys():
+    private_key_obj = rsa.generate_private_key(
+        backend=default_backend(),
+        public_exponent=65537,
+        key_size=2048
+    )
+    private_key = private_key_obj.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=serialization.NoEncryption()
+    )
+    public_key = private_key_obj.public_key().public_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PublicFormat.SubjectPublicKeyInfo
+    )
+    return private_key, public_key
+
+
+# Helper function to write the serverCaCert, ephemeral certificate and private
+# key to .pem files
+def write_to_file(serverCaCert, ephemeralCert, priv_key):
```

Comment:
Do these have to be in files? Can you use in memory ones instead?

Reply:
I think they do, the pymysql connector only takes filepaths as arguments. I'll ask Chris Wilcox or Thea Flowers to see if Python has a way to fake it somehow.

Reply:
Looks like they have to be in files.